pkg/command/command.go

```diff
@@ -7,100 +7,44 @@ import (
 	"os"
 
 	"github.com/jfrog/jfrog-client-go/utils/io/fileutils"
+
 	"github.com/xenitab/gitops-promotion/pkg/config"
 	"github.com/xenitab/gitops-promotion/pkg/git"
 )
 
-func getConfig(path string) (config.Config, error) {
-	file, err := os.Open(fmt.Sprintf("%s/gitops-promotion.yaml", path))
-	if err != nil {
-		return config.Config{}, err
-	}
-	cfg, err := config.LoadConfig(file)
-	if err != nil {
-		return config.Config{}, fmt.Errorf("could not load config: %w", err)
-	}
-	return cfg, nil
-}
+const (
+	configFileName = "gitops-promotion.yaml"
+)
 
-func getRepository(ctx context.Context, providerType string, path, token string) (*git.Repository, error) {
-	repo, err := git.LoadRepository(ctx, path, providerType, token)
-	if err != nil {
-		return nil, fmt.Errorf("could not load repository: %w", err)
+func Run(ctx context.Context, args []string) (string, error) {
+	// Global flags
+	if len(args) < 2 {
+		return "", fmt.Errorf("new, promote or status subcommand is required")
 	}
-	return repo, nil
-}
-
-func Run(args []string) (string, error) {
 	defaultPath, err := os.Getwd()
 	if err != nil {
 		return "", err
 	}
-
-	newCommand := flag.NewFlagSet("new", flag.ExitOnError)
-	newToken := newCommand.String("token", "", "Access token (PAT) to git provider")
-	newGroup := newCommand.String("group", "", "Main application group")
-	newApp := newCommand.String("app", "", "Name of the application")
-	newTag := newCommand.String("tag", "", "Application version/tag to set")
-	newProviderType := newCommand.String("provider", "azdo", "git provider to use")
-	newPath := newCommand.String("sourcedir", defaultPath, "Source working tree to operate on")
-
-	promoteCommand := flag.NewFlagSet("promote", flag.ExitOnError)
-	promoteToken := promoteCommand.String("token", "", "Access token (PAT) to git provider")
-	promoteProviderType := promoteCommand.String("provider", "azdo", "git provider to use")
-	promotePath := promoteCommand.String("sourcedir", defaultPath, "Source working tree to operate on")
-
-	statusCommand := flag.NewFlagSet("status", flag.ExitOnError)
-	statusToken := statusCommand.String("token", "", "Access token (PAT) to git provider")
-	statusProviderType := statusCommand.String("provider", "azdo", "git provider to use")
-	statusPath := statusCommand.String("sourcedir", defaultPath, "Source working tree to operate on")
-
-	if len(args) < 2 {
-		return "", fmt.Errorf("new, promote or status subcommand is required")
+	global := flag.NewFlagSet(args[1], flag.ExitOnError)
+	token := global.String("token", "", "Access token (PAT) to git provider")
+	providerType := global.String("provider", "azdo", "The git provider to use")
+	path := global.String("sourcedir", defaultPath, "Source working tree to operate on")
+	err = global.Parse(args[2:])
```

Comment:
From the failing test, it looks like this does not work? The parse operation will fail because it does not recognize -group.

Reply:
We run integration tests against the built container https://github.com/XenitAB/gitops-promotion/runs/4924366745?check_suite_focus=true#step:13:94. Try something like `env GITHUB_URL='https://github.com/gitops-promotion/gitops-promotion-testing' GITHUB_TOKEN=(cat github-token) GITOPS_PROMOTION_IMAGE=ghcr.io/xenitab/gitops-promotion:dev  go test -count=1 ./tests/...` to run on command line.